File: oop/heritage/Tp3/src/Employe.java
public class Employe extends Person {
    double salary;

    public Employe(String lastname, String firstname, double salary) {
        super(lastname, firstname);
        this.salary = salary;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    @Override
    public String toString() {
        return "Employe [salary=" + salary + ", Id=" + getId() + ", lastname=" + getLastname() + ", firstname=" + getFirstname()
                + "]";
    }
}

File: oop/heritage/CPP/exercice2/src/Employe.cpp
#include "Employe.hpp"

Employe::Employe(const std::string& lastname, const std::string& firstname, double salary)
    : Person(lastname, firstname), salary(salary) {}

double Employe::getSalary() const {
    return salary;
}

void Employe::setSalary(double newSalary) {
    salary = newSalary;
}

std::string Employe::toString() const {
    return Person::toString() + "Employe [salary=" + std::to_string(salary) + "]";
}
